### ISSUE TITLE
Services of tedge mapper and tedge agent fail when you reboot system after cumulocity connection

### ISSUE THREAD

**Issue by Ruadhri17** [thin-edge/meta-tedge#25]
When you reboot your system after first successful connection with a cumulocity cloud, the `tedge mapper` service along with `tedge agent` service will fail. 

To reproduce:

1. Run Yocto, connect with your c8y cloud.
2. Send exemplary telemetry data using `tedge/measurements`.
3. Reboot Yocto system.
4. After booting try again to send telemetry data using `tedge/measurements` (it should not work).
5. Check active services using `systemctl` command - it should prompt that `thin-edge` services failed. 

`Tedge agent` service gives following message after checking its status:
```bash
x tedge-agent.service - tedge-agent is a thin-edge.io component to support operations.
     Loaded: loaded (/lib/systemd/system/tedge-agent.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Fri 2022-12-02 12:55:30 UTC; 4min 13s ago
    Process: 239 ExecStart=/usr/bin/tedge_agent (code=exited, status=1/FAILURE)
   Main PID: 239 (code=exited, status=1/FAILURE)

Dec 02 12:55:30 qemuarm64 systemd[1]: tedge-agent.service: Scheduled restart job, restart counter is at 5.
Dec 02 12:55:30 qemuarm64 systemd[1]: Stopped tedge-agent is a thin-edge.io component to support operations..
Dec 02 12:55:30 qemuarm64 systemd[1]: tedge-agent.service: Start request repeated too quickly.
Dec 02 12:55:30 qemuarm64 systemd[1]: tedge-agent.service: Failed with result 'exit-code'.
Dec 02 12:55:30 qemuarm64 systemd[1]: Failed to start tedge-agent is a thin-edge.io component to support operations..
```
Similarly, `tedge mapper c8y` gives:
```bash
x tedge-mapper-c8y.service - tedge-mapper-c8y converts Thin Edge JSON measurements to Cumulocity JSON format.
     Loaded: loaded (/lib/systemd/system/tedge-mapper-c8y.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Fri 2022-12-02 12:55:31 UTC; 6min ago
    Process: 250 ExecStart=/usr/bin/tedge_mapper c8y (code=exited, status=1/FAILURE)
   Main PID: 250 (code=exited, status=1/FAILURE)

Dec 02 12:55:31 qemuarm64 systemd[1]: tedge-mapper-c8y.service: Scheduled restart job, restart counter is at 5.
Dec 02 12:55:31 qemuarm64 systemd[1]: Stopped tedge-mapper-c8y converts Thin Edge JSON measurements to Cumulocity JSON format..
Dec 02 12:55:31 qemuarm64 systemd[1]: tedge-mapper-c8y.service: Start request repeated too quickly.
Dec 02 12:55:31 qemuarm64 systemd[1]: tedge-mapper-c8y.service: Failed with result 'exit-code'.
Dec 02 12:55:31 qemuarm64 systemd[1]: Failed to start tedge-mapper-c8y converts Thin Edge JSON measurements to Cumulocity JSON format..
```

After further investigation, I think it may be caused by `/var/lock` permissions, they cannot create files there and as a result they try to restart the service, but fail every time. I will try to change the way we give permission to tedge user and see if that will help.

**Comment by Bravo555** [comment]
The issue happens because by default in Yocto, `/var/log/` directory is volatile, and is actually a symlink to `/var/volatile/log/`, with `/var/volatile` being a ramdisk. This makes it so when the postinst scripts run, and create a `/var/log/tedge/` directory, these changes are lost after the device is rebooted.

The quick and dirty solution for now is to add `VOLATILE_LOG_DIR = "no"` to your `build/conf/local.conf` file.

Possible long-term solutions are:

- Figure out how to enforce `VOLATILE_LOG_DIR = "no"` at the level of the layer.
  This fixes the issue, but enforces a choice on the user, who otherwise might not want to persist the logs (e.g. because they're heavily storage-limited)
- Change how the mapper is launched so that it has permissions to write to `/var/log/` anytime. E.g. mosquitto didn't have the problem with persisting the log dir, because it created one every boot
- **(preferable)** Remove the hard requirement for a logfile, by either issuing a warning when it can't be created and continuing, or by scrapping the file entirely and just logging everything using Rust's [log](https://lib.rs/crates/log) facade, which we already use

**Comment by Ruadhri17** [comment]
Problem was solved in PR #36. Now symlink is deleted between `/var/log` and `var/volatile/log` and logs are persistent. Other changes: now permissions for /run/lock are [patched](https://patchwork.yoctoproject.org/project/oe-core/patch/20220905122611.3957244-1-niko.mauno@vaisala.com/) in poky and there is no need to change them manually for Kirkstone distro. However, patch was not introduced for Honister, therefore there is a conditional that checks if currently used distro is Honister. If so, permissions are changed.
